src/Utility/MultipartFormDataBinaryContent.cs

```diff
@@ -120,27 +120,46 @@ public static void AddContentTypeHeader(HttpContent content, string contentType)
         content.Headers.ContentType = header;
     }
 
+#if NET
+    private static string CreateBoundary() =>
+        string.Create(BoundaryLength, 0, (chars, _) =>
+        {
+            Span<byte> random = stackalloc byte[BoundaryLength];
+            Random.Shared.NextBytes(random);
+
+            for (int i = 0; i < chars.Length; i++)
+            {
+                chars[i] = BoundaryValues[random[i] % BoundaryValues.Length];
+            }
+        });
+#else
+    private static readonly Random _random = new();
+
     private static string CreateBoundary()
     {
-        Span<char> chars = new char[70];
+        Span<char> chars = stackalloc char[BoundaryLength];
 
-        byte[] random = new byte[70];
-        _random.NextBytes(random);
+        byte[] random = new byte[BoundaryLength];
+        lock (_random)
+        {
+            _random.NextBytes(random);
+        }
 
         // The following will sample evenly from the possible values.
         // This is important to ensuring that the odds of creating a boundary
         // that occurs in any content part are astronomically small.
-        int mask = 255 >> 2;
+        const int Mask = 255 >> 2;
 
-        Debug.Assert(_boundaryValues.Length - 1 == mask);
+        Debug.Assert(BoundaryValues.Length - 1 == Mask);
 
-        for (int i = 0; i < 70; i++)
+        for (int i = 0; i < random.Length; i++)
```

Comment:
```suggestion
        for (int i = 0; i < BoundaryLength; i++)
```

Reply:
Another one of those "Steve is trying to microoptimize... do we care?" scenarios.

The .NET Framework JIT doesn't know how to remove the bounds check on BoundaryValues here (the .NET Core one could). There are three bounds checks possible here: one on indexing into chars, one on indexing into BoundaryValues, and one on indexing into random. As it was written before, none could be removed on netfx. By changing it to iterate over the full length of random, at least the bounds check from `random[i]` is removed.

But, it's really not a big deal. If you prefer the source code consistency, I can certainly change it.

Reply:
> Another one of those "Steve is trying to microoptimize... do we care?" scenarios.

@annelo-msft, I think Steve is giving you a green light to push for adding Variant to the BCL :-)

Reply:
No, we like optimizations, and thanks for explaining so we can learn!  If you wouldn't mind adding a comment to that effect so the next person who wonders the same thing can have their question answered inline in the code, then I think we're happy to have the optimization.  Many thanks!

Reply:
> If you wouldn't mind adding a comment to that effect so the next person who wonders the same thing can have their question answered inline in the code, then I think we're happy to have the optimization. Many thanks!

Okey dokey